gprofiler/utils/perf.py

```diff
@@ -3,14 +3,67 @@
 # Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
 #
 
-from gprofiler.exceptions import CalledProcessError
+from enum import Enum
+from pathlib import Path
+from threading import Event
+
+from gprofiler.exceptions import CalledProcessError, PerfNoSupportedEvent
 from gprofiler.log import get_logger_adapter
 from gprofiler.utils import run_process
-from gprofiler.utils.perf_process import perf_path
+from gprofiler.utils.perf_process import PerfProcess, perf_path
 
 logger = get_logger_adapter(__name__)
 
 
+class SUPPORTED_PERF_EVENTS(Enum):
+    PERF_DEFAULT = "default"
+    PERF_SW_CPU_CLOCK = "cpu-clock"
+    PERF_SW_TASK_CLOCK = "task-clock"
+
+    def perf_extra_args(self) -> list:
+        if self == SUPPORTED_PERF_EVENTS.PERF_DEFAULT:
+            return []
+        return ["-e", self.value]
+
+
+def perf_default_event_works(work_directory: Path, stop_event: Event) -> list:
```

Comment:
`storage_dir` is the term used across the project, otherwise use `tmp_dir`, don't add new terms

Reply:
do note we pass `storage_dir` for this function
but in the scope of this function it is indeed `tmp_dir`
544e75ff8883d6de5fc5ad339a79497e345b5863